syntax = "proto2";

import "ignition/msgs/vector2d.proto";
import "ignition/msgs/vector3d.proto";
import "xytheta.proto";

package smartmouse.msgs;

message RobotDescription {
    repeated ignition.msgs.Vector2d footprint = 1;
    optional WheelDescription left_wheel = 2;
    optional WheelDescription right_wheel = 3;
    optional ignition.msgs.Vector3d cog = 4;
    optional MotorDescription motor = 6;
    optional SensorsDescription sensors = 7;
}

message SensorsDescription {
    optional SensorDescription front = 1;
    optional SensorDescription gerald_left = 2;
    optional SensorDescription gerald_right = 3;
    optional SensorDescription front_left = 4;
    optional SensorDescription front_right = 5;
    optional SensorDescription back_left = 6;
    optional SensorDescription back_right = 7;
}

message SensorDescription {
    optional XYTheta p = 1;
    optional double min_range_m = 2;
    optional double max_range_m = 3;
    optional double beam_angle_rad = 6;
    optional double a = 7;
    optional double b = 8;
    optional double c = 9;
    optional double d = 10;
}

message MotorDescription {
    optional double u_kinetic = 1;
    optional double u_static = 2;
    optional double J = 3;
    optional double b = 4;
    optional double R = 5;
    optional double L = 6;
    optional double K = 7;
};

message WheelDescription {
   required ignition.msgs.Vector2d pose = 1;
   required double radius = 2;
   required double thickness = 3;
   required double u_static = 4;
}